### ISSUE TITLE
Rule CV12 deletes alias from table where columns are absent from the select statement

### ISSUE THREAD

**Issue by AndyBrownEJ** [sqlfluff/sqlfluff#6946]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Given a statement such as:
```
select a.a
from table_a as a
left join table_b as b
where a.id = b.id
```

noting that a) this triggers CV12 and b) no columns from table b exist in the select statement, using default values, sqlfluff fix incorrectly deletes the alias from table_b:

`sqlfluff fix --ignore-local-config --dialect ansi \flufftest\test1.sql`:

```
select a.a
from table_a as a
left join table_b on a.id = b.id
```

also of note, although `sqlfluff lint` shows only CV12 errors for ansi in the above statement, limiting fix to only rule CV12 performs correctly:

`pr sqlfluff fix --ignore-local-config --dialect ansi ..\flufftest\test1.sql -r CV12`:

```
select a.a
from table_a as a
left join table_b as b ON a.id = b.id
```

further, referencing table b in the select performs correctly:

```
select
    a.a,
    b.b
from table_a as a
left join table_b as b
where a.id = b.id
```

`sqlfluff fix --ignore-local-config --dialect ansi ..\flufftest\test1.sql`:

```
select
    a.a,
    b.b
from table_a as a
left join table_b as b on a.id = b.id
```

### Expected Behaviour

```
select a.a
from table_a as a
left join table_b as b on a.id = b.id
```

### Observed Behaviour

"as b" deleted:

```
select a.a
from table_a as a
left join table_b on a.id = b.id
```

### How to reproduce

step by step given in initial "what happened"

### Dialect

any or most (ansi used to demonstrate)

### Version

sqlfluff, version 3.4.0
python 3.10

### Configuration

default config applied, local config explicitly ignored

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by keraion** [comment]
Note: Conflict between AL05 and CV12
